test_util/helpers.py

```diff
@@ -211,3 +213,19 @@ def remove_file():
     # Attempt to remove the file upon normal interpreter exit.
     atexit.register(remove_file)
     return temp_path
+
+
+def lazy_property(property_fn):
```

Comment:
Would like to see some documentation on this function and also some tests to validate it's behavior since it's doing "fancy" things.